src/Azure/Orleans.Persistence.AzureStorage.Migration/Reminders/MigrationAzureTableReminderStorage.cs

```diff
@@ -0,0 +1,71 @@
+using Azure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Orleans.Configuration;
+using Orleans.Persistence.AzureStorage.Migration.Reminders.Storage;
+using Orleans.Persistence.Migration;
+using Orleans.Reminders.AzureStorage;
+using Orleans.Runtime;
+using Orleans.Runtime.ReminderService;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Orleans.Persistence.AzureStorage.Migration.Reminders
+{
+    /// <summary>
+    /// Simple storage provider for writing grain state data to Azure blob storage in JSON format.
+    /// Implementation impacts management of migrated and current data.
+    /// </summary>
+    public class MigrationAzureTableReminderStorage : IReminderTable
+    {
+        internal IReminderTable DefaultReminderTable { get; }
+        internal IReminderTable MigrationReminderTable { get; }
+
+        public MigrationAzureTableReminderStorage(
+            IGrainReferenceConverter grainReferenceConverter,
+            ILoggerFactory loggerFactory,
+            IGrainReferenceExtractor grainReferenceExtractor,
+            IOptions<ClusterOptions> clusterOptions,
+            IOptions<AzureTableReminderStorageOptions> oldStorageOptions,
+            IOptions<AzureTableMigrationReminderStorageOptions> migratedStorageOptions)
+        {
+            DefaultReminderTable = new AzureBasedReminderTable(grainReferenceConverter, loggerFactory, clusterOptions, oldStorageOptions);
+            MigrationReminderTable = new MigrationAzureBasedReminderTable(grainReferenceConverter, grainReferenceExtractor, loggerFactory, clusterOptions, migratedStorageOptions);
+        }
+
+        public async Task Init()
+        {
+            await MigrationReminderTable.Init();
+            await DefaultReminderTable.Init();
+        }
+
+        public Task<string> UpsertRow(ReminderEntry entry)
+        {
+            Task.Run(() => MigrationReminderTable.UpsertRow(entry)); // run in background is better here probably?
```

Comment:
I think it's too "dangerous" to not await the call

Reply:
Awaiting here. The initial idea was to not make app less performant by awaiting 2 upserts.